AsyncDisplayKit/ASRunLoopQueue.mm

```diff
@@ -235,24 +236,19 @@ - (void)processQueue
     ASProfilingSignpostStart(0, self);
 
     // Snatch the next batch of items.
-    NSUInteger totalNodeCount = _internalQueue.size();
-    for (int i = 0; i < MIN(self.batchSize, totalNodeCount); i++) {
-      id node = _internalQueue[0];
-      itemsToProcess.push_back(node);
-      _internalQueue.pop_front();
-    }
+    auto firstItemToProcess = _internalQueue.cbegin();
+    auto lastItemToProcess = MIN(_internalQueue.cend(), firstItemToProcess + self.batchSize);
+    itemsToProcess = std::vector<id>(firstItemToProcess, lastItemToProcess);
```

Comment:
I've confirmed it's safe to construct the iterator `firstItemToProcess + self.batchSize` even if that goes beyond the end of the queue. Tested by setting batch size to 10000 and then confirming that it safely grabbed the entire queue.